subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/DefaultTaskContainerTest.groovy

```diff
@@ -47,6 +48,15 @@ import static java.util.Collections.singletonMap
 
 class DefaultTaskContainerTest extends AbstractPolymorphicDomainObjectContainerSpec<Task> {
 
+    private taskCount = 1
+    private taskIdentityFactory = Mock(TaskIdentityFactory) {
```

Comment:
🤔 Would it make sense to use the real TaskIdentityFactory but mock the ConfigurationCacheableIdFactory instead?

Reply:
Addressed in another comment